### PR TITLE
FXIOS-18911 | Remove Deferred usage from Authenticator

### PR DESCRIPTION
**Pull request by imsanchit**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-18911)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18911)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/Authenticator.swift [hunk 1]
```diff
@@ -174,14 +204,14 @@ class Authenticator {
                 protectionSpace: protectionSpace
             )
                 self.sendLoginsAutofilledTelemetry()
-                deferred.fill(Maybe(success: login))
+                completionHandler(.success(login))
                 loginsHelper?.setCredentials(login)
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Browser/Authenticator.swift:208]: Indentation can be moved up a level here

File: firefox-ios/Client/Frontend/Browser/Authenticator.swift [hunk 2]
```diff
@@ -35,96 +40,120 @@ class Authenticator {
...
+                else if logins.count == 1 {
+                    credentials = logins[0].credentials
+                } else {
+                    logger.log("No logins found for Authenticator",
+                               level: .info,
+                               category: .webview)
+                }
 
-            return deferMaybe(credentials)
+                completionHandler(credentials)
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Browser/Authenticator.swift:153]: Recommend adding a space below 153 just for a little bit more readability, it's a big block to parse when reading

**Review by dataports — CHANGES_REQUESTED** [review]
hi there @imsanchit thanks for your contribution! I just have a couple readability suggestions but it looks great otherwise. When those are done I can approve and merge.

**Comment by imsanchit** [comment]
hi @dataports, have updated the PR as requested, please have a look.

**Review by dataports — APPROVED** [review]

**Pull request merged**